### PR TITLE
feat: implement ":contains" modifier

### PR DESCRIPTION
**Pull request by carvantes**
> [base]/Patient?given:contains=eve - Any  patients with a name with a given part containing "eve" at any  position. This would include patients with the given name "Eve",  "Evelyn", and also "Severine".

https://www.hl7.org/fhir/search.html#string

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ This package is dependent on:
 |---------------------|:----------:|-------|
 | `:missing`            |      -     |       |
 | `:exact`              |      -     |       |
```
Comment (ssvegaraju) [inline README.md]: this is a very small thing, but I didn't actually update the readme when implementing `exact:`, so maybe we could update that here?

**Review by ssvegaraju — APPROVED** [review]

File: src/QueryBuilder/typeQueries/stringQuery.ts [hunk 2]
```diff
@@ -10,13 +10,55 @@ const escapeQueryString = (string: string) => {
...
         throw new InvalidSearchParameterError(`Unsupported string search modifier: ${modifier}`);
     }
+
+    if (modifier === 'contains') {
+        const fields = [`${compiled.path}`];
+
+        if (compiled.path === 'name') {
+            // name is a special parameter.
+            // name.* cannot be used in wildcard queries so we specify the fields to search.
+            fields.push('name.family', 'name.given', 'name.text', 'name.prefix', 'name.suffix');
```
Comment (rsmayda) [inline src/QueryBuilder/typeQueries/stringQuery.ts:27]: Can we pull this from the IG compiler at all? (same for addresses)

Comment (carvantes) [reply src/QueryBuilder/typeQueries/stringQuery.ts:27]: Not really, the name param just has a text description that reads: 

> A  server defined search that may match any of the string fields in the  HumanName, including family, give, prefix, suffix, suffix, and/or text

File: README.md (hunk 1 above)
Comment (carvantes) [reply README.md]: yeah, I just updated it

**Review by rsmayda — APPROVED** [review]

**Pull request merged**